### PR TITLE
Fix FLG -pp ppx.exe -as-pp/-dump-ast on Windows

### PR DESCRIPTION
**Pull request by jonahbeckford**
Windows is special cased in Merlin to have a direct
 program execution without any shell so shell redirection
 does not work.

This is a narrow, Windows-only fix for
 https://github.com/ocaml/merlin/issues/1722

*I don't particularly like this one since it is based on heuristics.* And it does seem like there is a lot of unfixed code in other packages that is required to get `-as-pp` working on Windows. Perhaps someone more knowledgeable about Merlin can make this cleaner.

### CODE REVIEW

**Review by nojb — COMMENTED** [review]
Exactly what I had in mind, thanks!

**Review by nojb — APPROVED** [review]
LGTM

Someone with commit privileges will need to approve/merge (cc @voodoos)

File: src/utils/std.ml [hunk 1]
```diff
@@ -773,18 +773,21 @@ module System = struct
...
-             channel used by Merlin to communicate with the editor *)
-          if Sys.win32 then args else Format.sprintf "%s 1>&2" args
+      (* Runned program should never output on stdout since it is the
+          channel used by Merlin to communicate with the editor *)
+      let args =
+        if Sys.win32 then args
+        else
+          Printf.sprintf "%s 1>%s" args
+            (match stdout with | Some file -> Filename.quote file
+                               | None -> "&2")
```
Comment (voodoos) [inline src/utils/std.ml:783]:
```suggestion
          let stdout = match stdout with
            | Some file -> Filename.quote file
            | None -> "&2"
          in
          Printf.sprintf "%s 1>%s" args stdout
```

File: src/utils/std.ml [hunk 2]
```diff
@@ -773,18 +773,21 @@ module System = struct
...
       in
       let cmd = Format.sprintf "%s %s" prog args in
       let exit_code =
         if Sys.win32 then
-          (* Note: the following function will never output to stdout *)
-          windows_merlin_system_command cmd ~cwd
+          (* Note: the following function will never output to stdout
+             when [stdout = None]. Instead stdout is sent to stderr. *)
```
Comment (voodoos) [inline src/utils/std.ml:789]:
```suggestion
          (* Note: the following function will never output to stdout.
             When [stdout = None], stdout is sent to stderr. *)
```

**Review by voodoos — APPROVED** [review]
I am not expert enough to comment the Windows code, it looks all right and I will trust your judgement on it.

The rest of the PR looks fine to me, I just made a few stylistic propositions.

**Comment by nojb** [comment]
Friendly ping @voodoos: I think the PR can now be merged.

**Pull request merged**